src/components/PostsList.tsx

```diff
@@ -1,56 +1,57 @@
+/* eslint-disable @typescript-eslint/no-unused-expressions */
 import classNames from 'classnames';
 import React from 'react';
-import { Post } from '../types/Post';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { addPost, removePost } from '../features/selectedPost';
 
-type Props = {
-  posts: Post[],
-  selectedPostId?: number,
-  onPostSelected: (post: Post | null) => void,
-};
+export const PostsList: React.FC = () => {
+  const { items } = useAppSelector(state => state.posts);
+  const { item } = useAppSelector(state => state.selectedPost);
```

Comment:
It's better to rename it so it's understandable whet these variables store.
```suggestion
  const { items: posts } = useAppSelector(state => state.posts);
  const { item: selectedPost } = useAppSelector(state => state.selectedPost);
```